ci/unit/test_main_node.sh

```diff
@@ -3,7 +3,7 @@
 # This is a script to be run by the unit/test_main.sh to run a test
 # on a CI node.
 
-set -x
+set -uex
```

Comment:
This is a bit of a heavy hammer.  It's better in general to check return codes of each command within the script(s).  Sometimes they are not intended to be fatal.

Reply:
The general design that Brian and I have been using is that the -ue is always set, and a script should be protecting itself from anything that could return an error, and all variables should be defined or have a default.

This is to avoid the condition of a script failure going undetected causing a false positive for tests.

I will admit that I am learning more bash scripting techniques from dealing with  that.

Reply:
> This is a bit of a heavy hammer

It's not really a heavy hammer.  Because as you say:

> It's better in general to check return codes of each command within the script(s).

And that is **exactly** what `set -e` is there to ensure.  Or rather, at least stop things from continuing on blindly because the developer forgot/failed to add an error check where there was a failure.

>Sometimes they are not intended to be fatal.

That's perfectly fine.  But the developer really should acknowledge that with at minimum a `|| true`.   Then the reviewer and any subsequent readers and/or code modifiers know that the intention was that the command was being allowed to fail and not simply that the developer forgot to add error checking.

Same goes pretty much for `set -u`.  It prevents use of a variable the developer forgot to initialise.

You can think of these as the `bash` equivalent of `gcc`'s -Werror.